### PR TITLE
[SPARK-36393][BUILD][FOLLOW-UP] Try to raise memory for GHA

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

As followup to raise memory for two places forgotten.

### Why are the changes needed?

Raise memory for GHA.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

GA

### CODE REVIEW

**Comment by viirya** [comment]
cc @dongjoon-hyun @HyukjinKwon

**Comment by viirya** [comment]
cc @gengliangwang

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -1125,12 +1125,12 @@ object TestSettings {
     // SPARK-29282 This is for consistency between JDK8 and JDK11.
     (Test / javaOptions) ++= {
       val metaspaceSize = sys.env.get("METASPACE_SIZE").getOrElse("1300m")
-      s"-Xmx3200m -Xss4m -XX:MaxMetaspaceSize=$metaspaceSize -XX:+UseParallelGC -XX:-UseDynamicNumberOfGCThreads -XX:ReservedCodeCacheSize=128m"
+      s"-Xmx4g -Xss4m -XX:MaxMetaspaceSize=$metaspaceSize -XX:+UseParallelGC -XX:-UseDynamicNumberOfGCThreads -XX:ReservedCodeCacheSize=128m"
```
Comment (dongjoon-hyun) [inline project/SparkBuild.scala:1128]: Well, this is a global setting instead of GHA-only.

**Review by dongjoon-hyun — APPROVED** [review]

File: project/SparkBuild.scala (hunk 1 above)
Comment (viirya) [reply project/SparkBuild.scala:1128]: Yea, just set it back to previous level if GHA now allows.

**Comment by viirya** [comment]
GHA failed by

```
The downloaded source packages are in
	‘/tmp/RtmpqZ25tN/downloaded_packages’
Using bundled GitHub PAT. Please add your own PAT to the env var `GITHUB_PAT`
Error: Failed to install 'unknown package' from GitHub:
  HTTP error 401.
  Bad credentials
```

at "Install R linter dependencies and SparkR".

Looks unrelated.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by viirya** [comment]
Thanks @HyukjinKwon @dongjoon-hyun ! Merging to master/3.2.

**Pull request closed without merging**